### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by arunppsg**
In this PR, I am attempting some minor fixes to deepchem issues.

Some of the issues fixes are: #2362

### CODE REVIEW

**Comment by arunppsg** [comment]
Quite a long PR with lot of minor fixes. Here is the summary of changes introduced:

- [9bc2e1b](https://github.com/deepchem/deepchem/pull/2767/commits/9bc2e1b8cd6ab252b3b2ebc9130b628e99f72b8a) addresses change to the location of `import xgboost` - from being imported in `run_benchmark_models.py` to the specific function where it is required (`run_benchmark_models.xgboost`). The advantage is that it avoids the hard requirement of `xgboost` when running model benchmarks.
- [7739ce5](https://github.com/deepchem/deepchem/pull/2767/commits/7739ce5271a057d22c438611360b7f87b43ec826) improves `pdbbind_atomic_conv.py` example.
- [7deaa71](https://github.com/deepchem/deepchem/pull/2767/commits/7deaa712c03e331b393311275163faaacfad576a) raises an `Import Error` when the modules `mdtraj` and `pdbbindfixer` are not available for `AtomicConvFeaturizer`. This improves logging and for error traceback. [e99eca4](https://github.com/deepchem/deepchem/pull/2767/commits/e99eca45466ced5b563c05c6f1a4ba73394c6331) is related to the same.
- [8249152](https://github.com/deepchem/deepchem/pull/2767/commits/8249152f4d9cc657e375f08a5e72ed4e78dbd7da) and [d3a300c](https://github.com/deepchem/deepchem/pull/2767/commits/d3a300c34f0027d591057f4a2df360dc4546eb6b) adds flaky markers to tests.
- [1ff0810](https://github.com/deepchem/deepchem/pull/2767/commits/1ff081065ad299934854f16250039eca861a4ddc) removes unused imports.
- [30c92f10](https://github.com/deepchem/deepchem/pull/2767/commits/30c92f1006a60f3ce86d4357d54795b3b45a4d2d) is a minor fix in installation procedures. It is due to a mistake introduced in #2715 by me. Installation procedure for Darwin and other OS are different in `main.yml`. This commit restores it.
- [6787ee6](https://github.com/deepchem/deepchem/pull/2767/commits/6787ee6a8f92b69946b0d839ad09d1e125dd09b7) is an extension of #2732.

**Comment by arunppsg** [comment]
Regarding CI,

- Core: failing for ubuntu latest 3.9, windows latest 3.7
- Common: failing in windows latest 3.7
- Tensorflow, Torch, Jax: All passing

Note:
It is to be noted that some tests are still flaky (they passed in rerun). We have to track them in the future and add the `flaky` tags.
